nuxeo-core/nuxeo-core-test/src/test/java/org/nuxeo/ecm/core/TestColdStorage.java

```diff
@@ -109,6 +119,48 @@ public void shouldMoveToColdStorage() throws IOException {
         assertEquals("dummy", ((ManagedBlob) content).getProviderId());
     }
 
+    @Test
+    public void shouldMoveToColdStorageWhenUserHaveWriteColdStoragePermission() throws IOException {
+        DocumentModel documentModel = createFileDocument(DEFAULT_DOC_NAME, true);
+        ACP acp = documentModel.getACP();
+        ACL acl = acp.getOrCreateACL();
+        acl.add(new ACE("john", SecurityConstants.READ, true));
+        acl.add(new ACE("john", SecurityConstants.WRITE, true));
+        acl.add(new ACE("john", SecurityConstants.WRITE_COLD_STORAGE, true));
+        documentModel.setACP(acp, true);
+
```

Comment:
I am wondering if we can reduce this part of code and if we do wan reduce code on others parts. For example i was thinking if we can transform the `createFileDocument` to:
```
    protected DocumentModel createFileDocument(CoreSession session, boolean withBlobContent, ACE... aces) {
        DocumentModel documentModel = session.createDocumentModel("/", "MyFile", "File");
        if (withBlobContent) {
            documentModel.setPropertyValue(ColdStorageHelper.FILE_CONTENT_PROPERTY,
                    (Serializable) Blobs.createBlob(FILE_CONTENT));
        }
        DocumentModel document = session.createDocument(documentModel);
        if(aces.length > 0) {
            ACP acp = documentModel.getACP();
            ACL acl = acp.getOrCreateACL();
            acl.addAll(List.of(aces));
            document.setACP(acp, true);
        }
        return document;
    }
``` 

and the callers can be: 
`DocumentModel documentModel = createFileDocument(session, true, new ACE("john", SecurityConstants.READ, true));`
or
```
 ACE[] aces = { new ACE("john", SecurityConstants.READ, true), //
                new ACE("john", SecurityConstants.WRITE, true), //
                new ACE("john", SecurityConstants.WRITE_COLD_STORAGE, true) };

        DocumentModel documentModel = createFileDocument(session, true, aces);
```

Reply:
Yes, we can do it. 
Again, we'll duplicate similar logic on the following files:

- AbstractTestColdStorageOperation (https://github.com/nuxeo/nuxeo/blob/master/nuxeo-features/nuxeo-automation/nuxeo-automation-features/src/test/java/org/nuxeo/ecm/automation/core/operations/coldstorage/AbstractTestColdStorageOperation.java#L75-L82)
- TestColdStorage (https://github.com/nuxeo/nuxeo/blob/master/nuxeo-core/nuxeo-core-test/src/test/java/org/nuxeo/ecm/core/TestColdStorage.java#L241-L247)

Reply:
Ok! Let's factor the ACEs 👍